internal/bminventory/inventory.go

```diff
@@ -1696,15 +1696,11 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 	for _, h := range cluster.Hosts {
-		var host models.Host
-		var err error
-		if err = tx.Take(&host, "id = ? and cluster_id = ?",
-			h.ID.String(), cluster.ID.String()).Error; err != nil {
-			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
-				h.ID.String(), cluster.ID.String())
+		var host, err = getHost(h.ID.String(), cluster.ID.String(), tx, log)
```

Comment:
Is this code change required for this PR? If possible please isolate it to its own commit with its own message explaining why it changed. 

If it's not required, please add it to a separate PR

Reply:
done